test/functional/treesitter/parser_spec.lua

```diff
@@ -277,28 +277,6 @@ void ui_refresh(void)
     eq('void', res2)
   end)
 
-  it('support getting text where start of node is past EOF', function()
```

Comment:
@mfussenegger is this still something you think we should support. My intuition says if you pass an invalid node, you get an error.

Reply:
No strong opinion on it, I added this based on https://github.com/neovim/neovim/pull/14999#discussion_r666674419

Reply:
Ah ok. So basically we just need to cover the case when a range is zero width at EOF. That should be covered ok now whilst being stricter than before.